### PR TITLE
Feat/debt overview

### PR DESCRIPTION
**Pull request by clementbalestrat**

### CODE REVIEW

File: hooks/useHistoricalDebtData.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+					issuanceDebt: historicalIssuanceAggregation[i],
+					actualDebt: debtSnapshot.debtBalanceOf,
+				});
+			});
+
+			// Last occurrence is the current state of the debt
+			// Issuance debt = last occurrence of the historicalDebtAndIssuance array
+			historicalDebtAndIssuance.push({
+				timestamp: new Date().getTime(),
+				actualDebt: toBigNumber(debtDataQuery.data?.debtBalance ?? 0).toNumber(),
```
Comment (andytcf) [inline hooks/useHistoricalDebtData.ts:73]: don't convert to number, store as bigNumber and when displaying either use .toString() or convert to number then

File: hooks/useHistoricalDebtData.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+				issuanceDebt: last(historicalIssuanceAggregation) ?? 0,
+			});
+
+			setHistoricalDebt(historicalDebtAndIssuance);
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (andytcf) [inline hooks/useHistoricalDebtData.ts:79]: which deps are missing here, should prob include them and not to use eslint-disable anymore

File: pages/debt.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+const TotalSynthValue = styled(StatBox)`
+	.title {
+		color: ${(props) => props.theme.colors.purple};
+	}
+`;
+
+const Header = styled.div`
+	color: ${(props) => props.theme.colors.white};
+	font-family: ${(props) => props.theme.fonts.extended};
+	font-size: 16px;
```
Comment (andytcf) [inline pages/debt.tsx:100]: change font-size to 14px, i don't think we have 16 anywhere on the dApp

File: sections/debt/DebtChart/DebtChart.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+	);
+};
+
+type Data = {
+	issuanceDebt: number;
+	actualDebt: number;
+};
+
+const DebtChart = ({ data }: { data: Data[] }) => {
+	const { colors, fonts } = useContext(ThemeContext);
```
Comment (andytcf) [inline sections/debt/DebtChart/DebtChart.tsx:65]: ```const theme = useTheme``` would suffice as its apart of styled-components documentation


```import { useTheme } from 'styled-components';```

File: sections/debt/DebtChart/DebtChart.tsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+	if (!data || data.length === 0) return null;
```
Comment (andytcf) [inline sections/debt/DebtChart/DebtChart.tsx:66]: can we use a spinner for the edge case

```if(data.loading)```

**Review by andytcf — CHANGES_REQUESTED** [review]
Left some comments

File: pages/debt.tsx (hunk 3 above)
Comment (clementbalestrat) [reply pages/debt.tsx:117]: Wrong. 16px is used for infoBlock heading

File: sections/debt/DebtChart/DebtChart.tsx (hunk 5 above)
Comment (clementbalestrat) [reply sections/debt/DebtChart/DebtChart.tsx]: Good call

**Pull request merged**